app/models/base.py

```
@@ -82,40 +78,98 @@ def related_fields(self) -> List[str]:
+        return {
+            **fields,
+            **self._translate_related_fields(session, related_fields_data, create),
+        }
+
+    def _translate_related_fields(
+        self, session: Session, fields: dict, create: bool = False
+    ) -> dict:
         """Convert related fields from a dict to an instance of their declared model"""
         instances = {}
         for field_name, field_value in fields.items():
             field = getattr(self, field_name)
             if not field:
                 continue
             if isinstance(field, list):
-                model = field[0].model
+                request_class = field[0]
             else:
-                model = field.model
+                request_class = field
 
             if isinstance(field_value, list):
-                instances[field_name] = [model(**value) for value in field_value]
+                instances[field_name] = []
+                for value in field_value:
+                    instance = request_class.dict_to_instance(
+                        session, request_class.model, value, create
+                    )
+                    instances[field_name].append(instance)
             else:
-                instances[field_name] = model(**field_value)
+                instance = request_class.dict_to_instance(
+                    session, request_class.model, field_value, create
+                )
+                instances[field_name] = instance
         return instances
+
+    def dict_to_instance(
+        self, session: Session, model: SQLModel, values: dict, create: bool = False
+    ) -> SQLModel:
+        return model(**values)
+
+
+class BaseUpdateRequest(BaseRequest):
+    id: uuid.UUID | None = None
+
+    def dict_to_instance(
+        self, session: Session, model: SQLModel, values: dict, create: bool = False
+    ) -> SQLModel:
+        if not create and "id" in values:
+            instance = session.get(model, values["id"])
```

If this query returns `None` we should propagate this error back up rather than trying to call `sqlmodel_update(values)` on `None`.